commit 27488b3cd0e562533f364c8f0288c7923d69f4ad (refs/changes/06/4006/3)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-05-25T16:30:59+00:00 (1 year, 4 months ago)
    
    OS-6976 vmm_zsd_lock not held while element removed from vmm_zsd_list
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Bryan Cantrill <bryan@joyent.com>

diff --git a/usr/src/uts/i86pc/io/vmm/vmm_zsd.c b/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
index 586773ac8e..8313dfbb50 100644
--- a/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
+++ b/usr/src/uts/i86pc/io/vmm/vmm_zsd.c
@@ -154,6 +154,10 @@ vmm_zsd_destroy(zoneid_t zid, void *data)
 	vmm_zsd_t *zsd = data;
 	vmm_softc_t *sc;
 
+	mutex_enter(&vmm_zsd_lock);
+	list_remove(&vmm_zsd_list, zsd);
+	mutex_exit(&vmm_zsd_lock);
+
 	mutex_enter(&zsd->vz_lock);
 	ASSERT(!zsd->vz_active);
 
@@ -171,7 +175,6 @@ vmm_zsd_destroy(zoneid_t zid, void *data)
 	mutex_exit(&zsd->vz_lock);
 	mutex_destroy(&zsd->vz_lock);
 
-	list_remove(&vmm_zsd_list, zsd);
 	kmem_free(zsd, sizeof (*zsd));
 }
 
